Fix `pulsar --package` usage when `APM_PATH` is not set

savetheclocktower opened the pull request:
A user of `action-pulsar-dependency` reported this one. Note the output on the “Setup Pulsar Editor” step of [this CI job](https://github.com/jmanuel1/concat/actions/runs/20281302107/job/58243708141?pr=65):

```
Pulsar  : 1.130.2025120304
Electron: 30.0.9
Chrome  : 124.0.6367.233
Node    : 20.11.1
atom is not defined
ReferenceError: atom is not defined
    at PackageManager.getCommandName (C:\Users\runneradmin\AppData\Local\Programs\Pulsar\resources\app.asar\src\package-manager.js:182:26)
    at PackageManager.possibleApmPaths (C:\Users\runneradmin\AppData\Local\Programs\Pulsar\resources\app.asar\src\package-manager.js:192:30)
    at parseCommandLine (C:\Users\runneradmin\AppData\Local\Programs\Pulsar\resources\app.asar\src\main-process\parse-command-line.js:157:36)
    at start (C:\Users\runneradmin\AppData\Local\Programs\Pulsar\resources\app.asar\src\main-process\start.js:40:16)
    at Object.<anonymous> (C:\Users\runneradmin\AppData\Local\Programs\Pulsar\resources\app.asar\src\main-process\main.js:66:1)
    at Module._compile (node:internal/modules/cjs/loader:1391:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1451:10)
    at Module.load (node:internal/modules/cjs/loader:1214:32)
    at Module._load (node:internal/modules/cjs/loader:1030:12)
    at c._load (node:electron/js2c/node_init:2:13672)
```

That's an error after running `pulsar --package list`. And what a thoughtful and insightful stack trace we have!

## The cause (one of them, at least)

We parse `pulsar --package` as early as possible in order to minimize the amount of time we spend bootstrapping the editor in scenarios where the user is actually trying to run `ppm`. But in order to know the name of the binary we're meant to run (depends on the release channel!), we call a static method on `PackageManager`, and that method in turn calls a method defined on the `atom` global to figure out the right release channel. But the `atom` global isn't set up at this point!

The fix is easy: `atom.getReleaseChannel` just aliases a function that's defined elsewhere. We can import the original function directly instead of assuming `atom` will be defined.

## Why did it get this far?

We have main process code that envisions handling `--package` directly… but our wrapper scripts also look for `-p` and `--package` and try to delegate to `ppm` _before_ we run the main executable. So why are we hitting this code path in the first place?

I think [the wrong directory is being added to the `PATH`](https://github.com/pulsar-edit/action-pulsar-dependency/blob/main/action.yml#L47) in `action-pulsar-dependency`. We want the directory that contains `pulsar.sh` and `pulsar.cmd` to exist on the path so that a bare invocation of `pulsar` in a DOS/PowerShell session hits `pulsar.cmd` instead of `pulsar.exe`. So we should fix that in `action-pulsar-dependency`.

But this is why we still have code to handle `--package` in Pulsar itself — it's a useful fallback, and this is still a bug. Just makes it a bit hard to verify!

## Verification

On that note, here's how you can verify this fix.

First, to try to reproduce the bug: you can directly invoke the Pulsar binary — explicitly referencing the full path to `Pulsar.exe` on Windows:

```
C:\Path\To\Pulsar.exe --package list
```

That will probably trigger this bug.

If so, then the built version generated by this PR should have a fix. Once you install it, the same command ought to work just fine.

confused-Techie commented on src/package-manager.js:
```diff
@@ -8,6 +8,7 @@ const CSON = require('season');
 
 const ServiceHub = require('service-hub');
 const Package = require('./package');
+const { getReleaseChannel } = require('../get-app-details.js');
```
```suggestion
const { getReleaseChannel } = require('./get-app-details.js');
```

Seems this is what's causing tests to fail

savetheclocktower commented:
I think these CI failures are legit. I'll investigate when I get the time.

savetheclocktower commented:
OK, the commit history was _strange_ on this one. Rebased it from `master` and made it much less strange. Rebased it again to remove redundancy and then the strangeness returned, so I'm quitting while I'm ahead.

DeeDeeG commented:
With invitation from PR author, I've pushed a "no diff" cleanup of the rebasing using the method described in [my comment above](https://github.com/pulsar-edit/pulsar/pull/1388#issuecomment-4061984356).

The comparison of before / after indeed shows "There isn't anything to compare" ("no diff" in my phrasing): [`ac50fdaa7b`..`dcf598f0fd`](https://github.com/pulsar-edit/pulsar/compare/ac50fdaa7ba754ba0f0b07b3ae2c44c7c29bfe73..dcf598f0fdd98ee61840e2641c2960a0c6e60d72)

DeeDeeG reviewed (approved):
I confirm the original issue... that is: following the "Verification" section of the PR description, that **without** this change (that is, using the latest Rolling on Windows, extracted from .zip, run via full explicit path in cmd.exe) I **do** get the `ReferenceError: atom is not defined` symptom described in the "Verification" section of the PR description.

I confirm the fix... that is: with this PR's binary, I get a successful run of `pulsar --package list` AKA `ppm list`.

_(Side note: The output of `Pulsar.exe --package list` appears garbled due to newlines and color control characters not rendering as intended, but it runs. I would have to guess, but I'm _not_ at all inclined to guess this PR caused this formatting issue! If I instead do `C:\Path\To\ppm.cmd list`, the formatting is fine, blue coloring of the "Community Packages" line and all!)_

(and FWIW the fix worked the same for me when using a CI binary from an alternate branch without the `.js` extension thing (https://github.com/pulsar-edit/pulsar/pull/1388/changes/2040bbcb5ad70af60b0fb4fcd9f3118f1fd87d29) --> https://github.com/pulsar-edit/pulsar/actions/runs/23101530371. From discussion on Discord, I did not push that change to this branch, per @savetheclocktower's local testing, and to be on the safe side.)

Anyway, LGTM!
